### PR TITLE
Fix test failure on J9: do not mock jre class

### PR DESCRIPTION
**Pull request by jdyer1**
[No JIRA Issue]
- Fix test failure on J9: do not mock jre class

### CODE REVIEW

**Comment by jdyer1** [comment]
It has been a _Very long time_ since I have used my commit powers here.  This J9 test failure seemed like a nice way to try and get back into it a little.  I didn't test this with J9 but converting the mock on the jre class seemed like an easy way to fix this.  This now uses a real instance of the Map.  I will try and merge this in a day if there are no objections.

**Review by mkhludnev — APPROVED** [review]

**Review by uschindler — APPROVED** [review]

File: solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java
Comment (madrob) [inline solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java:193]: nit: two statements on this line

File: solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java
Comment (madrob) [inline solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java:161]: nit: could be assertThat(mapMock.get, isInstance()) for better error message

File: solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java
Comment (madrob) [inline solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java:74]: Can we clean up the assignments at line 51 and 54 since some of those are obsolete now I think?

**Review by madrob — APPROVED** [review]
Change looks good, left a few minor comments for more thorough cleanup.

**Comment by jdyer1** [comment]
Thank you @madrob for the review.  I cleaned a few things up, then pushed the commits via cherry-pick to "main" and "branch_9x".

**Comment by janhoy** [comment]
You left main branch failing precommit
```
Execution failed for task ':solr:core:spotlessJavaCheck'.
> The following files had format violations:
      src/test/org/apache/solr/core/BlobRepositoryMockingTest.java
```

**Comment by dsmiley** [comment]
@jdyer1 I strongly recommend against pushing commits to main from your machine.  If it were up to me, it'd be made to be impossible to help avoid mistakes.  Nowadays we use GitHub which runs some code quality/style checks that would have been in a failing state.  They would show on this PR screen here but are now absent because the state transitioned to closed.  Once you resolved those things, they would show in green and then you could merge via a Squash-Merge button.  It'd also ask for a commit message, which you should then edit.  Only the branch_9x backport would you do from your machine, but then only after waiting for a passing build on main.

**Comment by janhoy** [comment]
I fixed the formatting, but the test changes in this commit still fails errorprone tests

```bash
$ gradle solr:core:check -x test -Pvalidation.errorprone=true
> Task :solr:core:compileTestJava
/Users/janhoy/git/solr/solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java:117: warning: [DirectInvocationOnMock] Methods should not be directly invoked on the mock `mapMock`. Should this be part of a verify(..) call?
    assertTrue(mapMock.get("foo!") instanceof BlobRepository.BlobContent);
                          ^
    (see https://errorprone.info/bugpattern/DirectInvocationOnMock)
/Users/janhoy/git/solr/solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java:150: warning: [DirectInvocationOnMock] Methods should not be directly invoked on the mock `mapMock`. Should this be part of a verify(..) call?
    mapMock.put("foo!", new BlobRepository.BlobContent<BlobRepository>("foo!", blobData));
[...]
```

Perhaps you can revert and then fix in a new PR?

File: solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java
Comment (mkhludnev) [reply solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java:71]: Not getting your point. Please elaborate.

Comment (madrob) [reply solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java:71]: With this change, do we still use mocks field? Or is it used elsewhere and only skipped in this one test?

**Comment by mkhludnev** [comment]
pushed fix into main and branch_9x

File: solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java
Comment (mkhludnev) [reply solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java:71]: we don't have `mocks` field anymore https://github.com/apache/solr/blame/main/solr/core/src/test/org/apache/solr/core/BlobRepositoryMockingTest.java

**Pull request closed without merging**
